packages/twenty-front/src/modules/command-menu/hooks/useCommandMenu.ts

```diff
@@ -100,32 +107,110 @@ export const useCommandMenu = () => {
     [closeCommandMenu, openCommandMenu],
   );
 
-  const openRecordInCommandMenu = useRecoilCallback(
-    ({ set }) => {
-      return (recordId: string, objectNameSingular: string) => {
+  const navigateCommandMenu = useRecoilCallback(
+    ({ snapshot, set }) => {
+      return ({
+        page,
+        pageTitle,
+        pageIcon,
+      }: CommandMenuNavigationStackItem) => {
+        set(commandMenuPageState, page);
+        set(commandMenuPageInfoState, {
+          title: pageTitle,
+          Icon: pageIcon,
+        });
+
+        const currentNavigationStack = snapshot
+          .getLoadable(commandMenuNavigationStackState)
+          .getValue();
+
+        set(commandMenuNavigationStackState, [
+          ...currentNavigationStack,
+          { page, pageTitle, pageIcon },
+        ]);
         openCommandMenu();
-        set(commandMenuPageState, CommandMenuPages.ViewRecord);
-        set(viewableRecordNameSingularState, objectNameSingular);
-        set(viewableRecordIdState, recordId);
       };
     },
     [openCommandMenu],
   );
 
-  const openRecordsSearchPage = useRecoilCallback(
-    ({ set }) => {
+  const goBackFromCommandMenu = useRecoilCallback(
+    ({ snapshot, set }) => {
       return () => {
-        set(commandMenuPageState, CommandMenuPages.SearchRecords);
+        const currentNavigationStack = snapshot
+          .getLoadable(commandMenuNavigationStackState)
+          .getValue();
+
+        const newNavigationStack = currentNavigationStack.slice(0, -1);
+
+        if (newNavigationStack.length === 0) {
+          closeCommandMenu();
+        }
+
+        const lastNavigationStackItem =
+          newNavigationStack[newNavigationStack.length - 1];
```

Comment:
I like to use the new `Array.prototype.at()` function to get the last element:

```suggestion
          newNavigationStack.at(-1);
```

Reply:
Indeed !